src/test/run-pass/issue-18804/main.rs

```diff
@@ -0,0 +1,20 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Test for issue #18804, #[linkage] does not propagate thorugh generic
+// functions. Failure results in a linker error.
+
+// aux-build:lib.rs
```

Comment:
doesn't this file also need the `//ignore-*` comments? Or is compiletest smart enough to figure out that if an aux-build isn't built we also shouldn't run the test depending on it?

r=me with that question resolved

Reply:
You are correct. I put the comments in the wrong file while not thinking. The main test file needs the comments while the auxiliary file does not.